src/rubicon/objc/api.py

```diff
@@ -1717,40 +1862,252 @@ def __new__(cls, name_or_ptr):
+    lists that are created each time the methods are
+    called, which can have an effect on performance and memory usage for
+    large dictionaries. To avoid this, you can cache the return values of
+    [`keys`](https://docs.python.org/3/library/stdtypes.html#dict.keys) and
+    [`values`](https://docs.python.org/3/library/stdtypes.html#dict.values), or convert
+    the [`NSDictionary`][rubicon.objc.api.NSDictionary] to a Python [`dict`][]
+    beforehand.
+
+    [`items`](https://docs.python.org/3/library/stdtypes.html#dict.items) is currently
+    implemented as a generator, meaning that it returns a single-use iterator. If you
+    need to iterate over
+    [`items`](https://docs.python.org/3/library/stdtypes.html#dict.items) more than
+    once or perform other operations on it, you should convert it to a Python
+    [`set`][] or [`list`][] first.
+    """
+    NSMutableDictionary = ObjCClass("NSMutableDictionary")
+    """
+    The
+    [NSMutableDictionary](https://developer.apple.com/documentation/foundation/nsmutabledictionary?language=objc)
+    class from `<Foundation/NSDictionary.h>`.
+
+    This class can be converted to and from Python exactly like its
+    superclass `NSDictionary`.
+
+    Supports
+    [Python-style mutable mapping operations](https://docs.python.org/3/library/stdtypes.html#mapping-types-dict)
+    including `setitem()`, `delitem()`, `clear()`, `pop()`, `popitem()`, `setdefault()`,
+    and `update()`.
+    """
+    Protocol = ObjCClass("Protocol")
+    """
+    The
+    [Protocol](https://developer.apple.com/documentation/objectivec/protocol?language=objc)
+    class from `<objc/Protocol.h>`.
+
+    This class has no (non-deprecated) Objective-C methods; protocol objects
+    can only be manipulated using Objective-C runtime functions. Rubicon
+    automatically wraps all [`Protocol`][rubicon.objc.api.Protocol]
+    objects using [`ObjCProtocol`][rubicon.objc.api.ObjCProtocol], which
+    provides an easier interface for working with protocols.
+    """
 
 
 def py_from_ns(nsobj):
     """Convert a Foundation object into an equivalent Python object if possible.
 
     Currently supported types:
 
-    * :class:`~rubicon.objc.runtime.objc_id`: Wrapped in an
-         :class:`ObjCInstance` and converted as below
-    * :class:`NSString`: Converted to :class:`str`
-    * :class:`NSData`: Converted to :class:`bytes`
-    * :class:`NSDecimalNumber`: Converted to :class:`decimal.Decimal`
-    * :class:`NSDictionary`: Converted to :class:`dict`, with all keys and
-         values converted recursively
-    * :class:`NSArray`: Converted to :class:`list`, with all elements converted
-         recursively
-    * :class:`NSNumber`: Converted to a :class:`bool`, :class:`int` or
-         :class:`float` based on the type of its contents
-
-    Other objects are returned unmodified as an :class:`ObjCInstance`.
+    * [`objc_id`][rubicon.objc.runtime.objc_id]: Wrapped in an
```

Comment:
Some odd indentation and bolding here (and in `ns_for_py` - I think it's the newline being interpreted as a new bullet level?